.github/workflows/causal_lm_cpp.yml

```
@@ -349,6 +349,55 @@ jobs:
           && timeout 50s ./build/samples/cpp/beam_search_causal_lm/beam_search_causal_lm ./notus-7b-v1/ 69
           | diff <(timeout 50s ./samples/python/beam_search_causal_lm/beam_search_causal_lm.py ./notus-7b-v1/ 69) -
 
+  cpp-beam_search_causal_lm-MiniCPM-2B-dpo:
+    runs-on: ubuntu-20.04-16-cores
```

```suggestion
    runs-on: ubuntu-20.04
```
Try a smaller runner. If it fails, switch to `ubuntu-20.04-4-cores`